mgmt_cli_test.py

```diff
@@ -506,6 +512,86 @@ def test_backup_rate_limit(self):
         self.verify_backup_success(mgr_cluster=mgr_cluster, backup_task=backup_task)
         self.log.info('finishing test_backup_rate_limit')
 
+    @staticmethod
+    def _get_all_snapshot_files_s3(cluster_id, bucket_name, region_name):
+        file_set = set()
+        try:
+            s3_client = boto3.client('s3', region_name=region_name)
+            paginator = s3_client.get_paginator('list_objects')
+            pages = paginator.paginate(Bucket=bucket_name, Prefix=f'backup/sst/cluster/{cluster_id}')
+            for page in pages:
+                content_list = page["Contents"]
+                file_set.update([item["Key"] for item in content_list])
+        except KeyError as err:
+            if err.args[0] == "Contents":
+                pass  # No snapshot files of the cluster exist, probably no backup ran before this function
+            else:
+                raise
+        return file_set
+
+    @staticmethod
+    def _get_all_snapshot_files_gce(cluster_id, bucket_name):
+        file_set = set()
+        gcp_credentials = KeyStore().get_gcp_credentials()
+        gce_driver = libcloud.storage.providers.get_driver(libcloud.storage.types.Provider.GOOGLE_STORAGE)
+        driver = gce_driver(gcp_credentials["project_id"] + "@appspot.gserviceaccount.com",
+                            gcp_credentials["private_key"],
+                            project=gcp_credentials["project_id"])
+        container = driver.get_container(container_name=bucket_name)
+        dir_listing = driver.list_container_objects(container, ex_prefix=f'backup/sst/cluster/{cluster_id}')
```

Comment:
perhaps a paginator relevant here as well, similar to aws?

Reply:
No need, it isn't required with libcloud